dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -292,6 +302,12 @@ await foreach (var chatCompletionStreamingResult in chatCompletionStreamingResul
     private static OpenAIChatHistory PrepareChatHistory(string text, AIRequestSettings? requestSettings, out OpenAIRequestSettings settings)
     {
         settings = OpenAIRequestSettings.FromRequestSettings(requestSettings);
+
+        if (ChatPromptParser.TryGetChatHistory(text, out var chatHistory) && chatHistory is not null)
```

Comment:
I think the additional check of the `chatHsitory` for null is not necessary here. The `TryGetChatHistory` should always initialize chat history if it returns true.